src/main.ts

```diff
@@ -1161,6 +1106,78 @@ export default class RemotelySavePlugin extends Plugin {
 
     this.syncOnRemoteIntervalID = window.setInterval(syncOnRemote, this.settings.syncOnRemoteChangesAfterMilliseconds);
   }
+
+  async toggleSyncOnSave(enabled: boolean) {
+    let alreadyScheduled = false;
+
+    // Unregister vault change event
+    if (this.syncOnSaveEvent !== undefined) {
+      this.app.vault.offref(this.syncOnSaveEvent);
+      this.syncOnSaveEvent = undefined;
+    }
+
+    // Unregister scanning for .obsidian changes
+    if (this.vaultScannerIntervalId !== undefined) {
+      window.clearInterval(this.vaultScannerIntervalId);
+      this.vaultScannerIntervalId = undefined;
+    }
+
+    if (enabled === false || this.settings.syncOnSaveAfterMilliseconds === -1) {
+      return;
+    }
+    
+    // Register vault change event
+    this.syncOnSaveEvent = this.app.vault.on("modify", () => {
+      if (this.syncStatus !== "idle" || alreadyScheduled) {
+        return;
+      }
+
+      alreadyScheduled = true;
+      log.debug(`Scheduled a sync run for ${this.settings.syncOnSaveAfterMilliseconds} milliseconds later`);
+
+      setTimeout(async () => {
+        log.debug("Sync on save ran");
+        await this.syncRun("auto");  
+        alreadyScheduled = false;
+      }, this.settings.syncOnSaveAfterMilliseconds);
+    });
+
+    // Scan vault for config directory changes
+    const scanVault = async () => {
+      if (this.syncStatus !== "idle" || alreadyScheduled || !this.settings.syncConfigDir) {
+        return;
+      }
+
+      log.debug("Scanning config directory for changes");
+
+      let localConfigContents: ObsConfigDirFileType[] = await listFilesInObsFolder(this.app.vault, this.manifest.id, this.settings.syncTrash);
+
+      for (let i = 0; i < localConfigContents.length; i++) {
+        const file = localConfigContents[i];
+
+        if (file.key.includes(".obsidian/plugins/remotely-sync/")) {
```

Comment:
This should be remotely-secure right?

Reply:
Yes sorry I think it should be. The reason I had remotely-sync was because of my local repository name. This should probably be fixed as I think the data.json gets modified each sync, resulting in the vault scan thinking there's a file change.